tartiflette/sdl/schema.py

```diff
@@ -0,0 +1,285 @@
+        # All non-abstract types possible for a given abstract type
+        self._possible_types: Dict[str, Dict[str, bool]] = {}
+
+    def __repr__(self):
+        return "GraphQLSchema(query: {}, " \
+               "mutation: {}, " \
+               "subscription: {}, " \
+               "types: {}, " \
+               "directives: {})".format(
+                self._query_type,
+                self._mutation_type,
+                self.subscription_type,
+                self._gql_types,
+                self._directives,
+                )
+
+    def __eq__(self, other):
+        if self._query_type != other._query_type or \
+            self._mutation_type != other._mutation_type or \
+            self._subscription_type != other._subscription_type:
+            return False
+        if len(self._gql_types) != len(other._gql_types):
+            return False
+        for key in self._gql_types:
+            if key not in other._gql_types:
+                return False
+            elif self._gql_types[key] != other._gql_types[key]:
+                return False
+        return True
+
+    @property
+    def query_type(self) -> Optional[GraphQLNamedType]:
+        return self._query_type
+
+    @query_type.setter
+    def query_type(self, value: GraphQLNamedType) -> None:
+        if not isinstance(value, GraphQLNamedType):
+            raise ValueError(
+                'Schema `query` must be a '
+                '`GraphQLNamedType`, got `{}`'.format(
+                    value.__class__.__name__
+                )
+            )
+        self._query_type = value
+
+    @property
+    def mutation_type(self) -> Optional[GraphQLNamedType]:
+        return self._mutation_type
+
+    @mutation_type.setter
+    def mutation_type(self, value: GraphQLNamedType) -> None:
+        if not isinstance(value, GraphQLNamedType):
+            raise ValueError(
+                'Schema `mutation` must be a '
+                '`GraphQLNamedType`, got `{}`'.format(
+                    value.__class__.__name__
+                )
+            )
+        self._mutation_type = value
+
+    @property
+    def subscription_type(self) -> Optional[GraphQLNamedType]:
+        return self._subscription_type
+
+    @subscription_type.setter
+    def subscription_type(self, value: GraphQLNamedType) -> None:
+        if not isinstance(value, GraphQLNamedType):
+            raise ValueError(
+                'Schema `subscription` must be a '
+                '`GraphQLNamedType`, got `{}`'.format(
+                    value.__class__.__name__
+                )
+            )
+        self._subscription_type = value
+
+    def add_definition(self, value: GraphQLNamedTypeDefinition) -> None:
+        # TODO: Check stuff, call update_schema after each new definition ?
+        self._gql_types[value.name] = value
+
+    def get_field_by_name(self, name: str) -> Optional[GraphQLFieldDefinition]:
```

Comment:
name can be either "the field name" or "a path" like .e.g "User.name", I'm wondering if "name" is the most suited name.

Reply:
The path is the fully qualified name `Object.fieldName` so I think it's OK. That won't be the case for the get_resolver of course which can take a full "path" `fieldName.subField.subSubField`.